files/scripts/syncd.sh

```diff
@@ -21,7 +21,27 @@ function startplatform() {
     # start mellanox drivers regardless of
     # boot type
     if [[ x"$sonic_asic_platform" == x"mellanox" ]]; then
-        BOOT_TYPE=`getBootType`
+        metadata=$(sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]')
```

Comment:
Avoid calling sonic-cfggen at boot since it is a heavy command.
Use this approach https://github.com/sonic-net/sonic-buildimage/blob/master/files/image_config/config-chassisdb/config-chassisdb#L30